cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -168,20 +161,28 @@ protected void initGoalState(ClusterModel clusterModel, Set<String> excludedTopi
    */
   @Override
   protected void updateGoalState(ClusterModel clusterModel, Set<String> excludedTopics) throws AnalysisInputException {
+    Set<Integer> brokerIdsAboveBalanceUpperLimit = new HashSet<>();
+    Set<Integer> brokerIdsUnderBalanceLowerLimit = new HashSet<>();
     // Log broker Ids over balancing limit.
     // While proposals exclude the excludedTopics, the balance still considers utilization of the excludedTopic replicas.
-    if (!_brokerIdsAboveBalanceUpperLimit.isEmpty()) {
+    for (Broker broker : clusterModel.healthyBrokers()) {
+      if (!isLoadUnderBalanceUpperLimitAfterChange(clusterModel, null, broker, REMOVE)) {
```

Comment:
When `isLoadUnderBalanceUpperLimitAfterChange(...)` is called with `null` load, it checks whether the current load is under the balance upper limit or not. Hence, it is a little bit confusing to pass `REMOVE` as the `changeType` argument, which has no effect in the way this function works in this case. Can we maybe pass `null` as the change type to make it more clear that change is irrelevant? Or would a new function that specifically performs this check is more clear? 
-- We use this function the same way in one more place.